File: client/src/dataModels/company.ts
import { BaseEntityModel } from "./baseEntityModel";

export class Company extends BaseEntityModel {
    Name: string;
    Address: string;
    SecondaryAddress: string;
    City: string;
    State: string;
    ZipCode: string;
    Phone: string;
    Fax: string;
    WebSiteUrl: string;
    PatientDataModelId: string;

    constructor(id: string = "", isDelete: boolean = false) {
        super(id, isDelete);

        this.Name = "";
        this.Address = "";
        this.SecondaryAddress = "";
        this.City = "";
        this.State = "";
        this.ZipCode = "";
        this.Phone = "";
        this.Fax = "";
        this.WebSiteUrl = "";
        this.PatientDataModelId = "";

    }

    createFromEntityModel(companyModel: any) {
        const company =
            new Company(companyModel.Id, !!companyModel.IsDelete);

        company.Name = companyModel.Name;
        company.Address = companyModel.Address;
        company.SecondaryAddress = companyModel.SecondaryAddress;
        company.City = companyModel.City;
        company.State = companyModel.State;
        company.ZipCode = companyModel.ZipCode;
        company.Phone = companyModel.Phone;
        company.Fax = companyModel.Fax;
        company.WebSiteUrl = companyModel.WebSiteUrl;
        company.PatientDataModelId = companyModel.PatientDataModelId;

        return company;
    }
}

File: medico/Domain/Medico.Domain/Models/Company.cs
using System.Collections.Generic;
using Medico.Domain.Enums;

namespace Medico.Domain.Models
{
    public class Company : Entity
    {
        public string Name { get; set; }

        public string Address { get; set; }

        public string SecondaryAddress { get; set; }

        public string City { get; set; }

        public int State { get; set; }

        public string ZipCode { get; set; }

        public ZipCodeType ZipCodeType { get; set; }

        public string Phone { get; set; }

        public string Fax { get; set; }

        public string WebSiteUrl { get; set; }

        public bool IsActive { get; set; }

        public List<Location> Locations { get; set; }

        public List<SelectableListCategory> SelectableListCategories { get; set; }

        public List<MedicoApplicationUser> MedicoApplicationUsers { get; set; }

        public List<SelectableList> SelectableLists { get; set; }

        public List<TemplateType> TemplateTypes { get; set; }

        public List<Patient> Patients { get; set; }

        public List<Appointment> Appointments { get; set; }

        public List<Phrase> Phrases { get; set; }

        public List<Template> Templates { get; set; }

        public List<PatientChartDocumentNode> PatientChartNodeDocuments { get; set; }
        
        public List<ReferenceTable> ReferenceTables { get; set; }
        
        public List<Expression> Expressions { get; set; }
    }
}